internal/codegen/stencil_test.go

```diff
@@ -126,6 +128,40 @@ func TestDirReplacementRendering(t *testing.T) {
 	assert.Equal(t, tps[0].Files[0].path, "bob/d/m1")
 }
 
+func TestBinaryRender(t *testing.T) {
+	log := slogext.NewTestLogger(t)
+	sm := &configuration.Manifest{Name: "testing", Arguments: map[string]any{"x": "d"}}
+	m1man := &configuration.TemplateRepositoryManifest{
+		Name: "testing1",
+	}
+	m1, err := modulestest.NewModuleFromTemplates(m1man, "testdata/binary.tplb")
+	assert.NilError(t, err, "failed to NewWithFS")
+
+	st := NewStencil(sm, nil, []*modules.Module{m1}, log, false)
+
+	tpls, err := st.Render(context.Background(), log)
+	assert.NilError(t, err, "failed to render template")
+	assert.Equal(t, len(tpls), 1)
+	assert.Equal(t, len(tpls[0].Files), 1)
+
+	wd, err := os.Getwd()
+	assert.NilError(t, err, "failed to get working directory")
+	td := os.TempDir()
+	err = os.Chdir(td)
```

Comment:
Is t.Chdir out yet? Can't wait to stop doing this...

Reply:
negative :(